app/helpers/samples_helper.rb

```diff
@@ -63,7 +63,47 @@ def parsed_samples_for_s3_path(s3_path, project_id, host_genome_id)
         sample_list << sample_attributes
       end
     end
-
     sample_list
   end
+
+  def samples_output_data(samples)
+    final_result = []
+    samples.each do |output|
+      output_data = {}
+      pipeline_info = output.pipeline_runs.first ? output.pipeline_runs.first.pipeline_output : nil
+      job_stats = output.pipeline_outputs.first ? output.pipeline_outputs.first.job_stats : nil
```

Comment:
I think this is a bug. This should be pipelin_info? Pipelin_info. Job_stats?
Also, why do you call it pipeline_info? Pipeline_output will be more informative.

Reply:
Agreed. Also, when you change `job_stats` to `job_stats = pipeline_info ? pipeline_info.job_stats : nil`, please make the same change in the corresponding line in `home_controller.rb`. (It was me who put the buggy line there in the first place...)

Reply:
sure @cdebourcy

Reply:
I actually moved the method into the sample helper so home controller does not have that line of code anymore @cdebourcy